source/Octopus.Client/LocationChecker.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Win32;
+using Octopus.Client.Logging;
+
+namespace Octopus.Client
+{
+#if FULL_FRAMEWORK
+    class LocationChecker
+    {
+        private static readonly ILog Logger = LogProvider.For<LocationChecker>();
+
+        public static void CheckAssemblyLocation()
+        {
+            var octopusRegNode = Registry.LocalMachine.OpenSubKey(@"Software\Octopus");
+            var serverRegNode = octopusRegNode.OpenSubKey("Server");
```

Comment:
You decided against worrying about Tentacle?

Reply:
I did. Could Tentacle be calling it from the install folder from inside PowerShell? 🤔 Would this happen if you ran a script step on a worker?

Reply:
I think the only way this would happen would be if it was a customer script, so we'd want the warning there too

Reply:
Updated to include warning for Tentacle too.